parla/task_runtime.py

```diff
@@ -2115,6 +2129,20 @@ def _launch_tasks(self):
                         if is_cpu or num_launched_datamove_task_count < (self._num_colocatable_tasks + 1):
                             self._launch_task(datamove_queue, dev, is_cpu, num_launched_datamove_task_count)
 
+    def start_scheduler_callbacks(self):
+        map_succeed = True
+        schedule_succeed = True
+        launch_succeed = True
+        map_succeed = self.map_tasks_callback()
+        schedule_succeed = self.schedule_tasks_callback()
+        launch_succeed = self.launch_tasks_callback()
+        while self.no_running_tasks():
+            if not map_succeed and not schedule_succeed and not launch_succeed:
```

Comment:
Commenting to check if this check is necessary/correct? e.g. is L2139 sufficient
- Are there edge cases where we'd want this loop more than once (assuming all phases execute)?

Reply:
That is for avoiding too many unncessary threads to loop over this main loop. I think only thread who blocked other threads needs to iterate.